dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/DefaultIdentifiableObjectManager.java

```diff
@@ -544,6 +548,23 @@ public void delete( IdentifiableObject object, User user )
         return (List<T>) store.getAllByAttributes( attributes );
     }
 
+    @Override
+    @Transactional( readOnly = true)
+    public <T extends IdentifiableObject> List<String> getAllValuesByAttributes( Class<T> klass, List<Attribute> attributes )
+    {
+        List<T> objects = getAllByAttributes( klass, attributes );
+        Set<String> attributeIds = attributes.stream().map( attribute -> attribute.getUid() )
```

Comment:
Is there a way to make this more effective by utilizing a database query?

Reply:
Fixed by using double query